plugins/retry/src/retry.spec.ts

```diff
@@ -1,7 +1,283 @@
-import { describe, expect, it } from 'vitest'
+/**
+ * @vitest-environment happy-dom
+ */
+import { enableAutoUnmount, flushPromises, mount } from '@vue/test-utils'
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+import { defineComponent } from 'vue'
+import { createPinia } from 'pinia'
+import { useQuery, PiniaColada } from '@pinia/colada'
+import type { UseQueryOptions } from '@pinia/colada'
+import { PiniaColadaRetry } from '.'
+import type { RetryOptions } from '.'
 
 describe('Pinia Colada Retry Plugin', () => {
-  it('works', () => {
-    expect(true).toBe(true)
+  beforeEach(() => {
+    vi.clearAllTimers()
+    vi.useFakeTimers()
+  })
+  afterEach(() => {
+    vi.restoreAllMocks()
+  })
+
+  enableAutoUnmount(afterEach)
+
+  const RETRY_OPTIONS_DEFAULTS = {
+    retry: 1,
+    delay: 500,
+  } as const
+
+  function factory(queryOptions: UseQueryOptions) {
+    const wrapper = mount(
+      defineComponent({
+        template: '<div></div>',
+        setup() {
+          return useQuery(queryOptions)
+        },
+      }),
+      {
+        global: {
+          plugins: [
+            createPinia(),
+            [
+              PiniaColada,
+              { plugins: [PiniaColadaRetry(RETRY_OPTIONS_DEFAULTS)] },
+            ],
+          ],
+        },
+      },
+    )
+
+    return { wrapper }
+  }
+
+  it('apply global retry options', async () => {
+    const query = vi.fn(async () => {
+      throw new Error('ko')
+    })
+
+    factory({
+      key: ['key'],
+      query,
+    })
+
+    // initial fetch fails
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(1)
+    // first retry
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+    // no second retry because the default retry is 1
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+  })
+
+  it('retry option with a number', async () => {
+    const query = vi.fn(async () => {
+      throw new Error('ko')
+    })
+
+    factory({
+      key: ['key'],
+      query,
+      retry: 2,
+    })
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(1)
+    // first retry
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+    // second retry
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(3)
+    // no further retries
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(3)
+  })
+
+  it('retry option with a function', async () => {
+    const query = vi.fn(async () => {
+      throw new Error('ko')
+    })
+    // stop retry when the error message is "ko"
+    const retry: RetryOptions['retry'] = (attempt, error) => {
```

Comment:
@coderabbitai can you inline this `retry` function into the option with the same name passed to `factory()`

Reply:
can you create a PR to this PR with all the changes you replied in this review?